File: content/atlas/source/includes/fts/synonyms/CreateDynamicIndex.java
import com.mongodb.client.*;
import org.bson.Document;

import java.util.List;

public class CreateDynamicIndex {
    public static void main(String[] args) {
        // Replace the placeholder with your MongoDB deployment's connection string
        String connectionString = "<connection-string>";

        try (MongoClient mongoClient = MongoClients.create(connectionString)) {
            MongoDatabase database = mongoClient.getDatabase("sample_mflix");
            MongoCollection<Document> collection = database.getCollection("movies");
            String indexName = "default";

            // Define the synonym mapping for the lucene.standard analyzer
            List<Document> synonymsList = List.of(
                    new Document("analyzer", "lucene.standard")
                            .append("name", "my_synonyms")
                            .append("source", new Document("collection", "synonymous_terms"))
            );

            // Create the MongoDB Search index definition
            Document searchIdx = new Document(
                    "mappings",
                    new Document("dynamic", true)
            ).append("synonyms", synonymsList);

            // Create the index by using the Java Sync Driver
            collection.createSearchIndex(indexName, searchIdx);
            System.out.println("New index name: " + indexName);
        } catch (Exception e) {
            System.err.println("Error creating Atlas Search index: " + e.getMessage());
        }
    }
}

File: content/search/source/includes/index/synonyms/code-snippets/nodejs/create-dynamic-index.js
const { MongoClient } = require("mongodb");

// Replace the connection string with your MongoDB deployment's connection string
const uri =  "<connection-string>";

const client = new MongoClient(uri);

async function run() {
  try {
    const database = client.db("sample_mflix");
    const collection = database.collection("movies");

    // Create the MongoDB Search index definition with dynamic mapping and synonyms
    const index = {
        name: "default",
        definition: {
            "mappings": {
                "dynamic": true
            },
            "synonyms": [
                {
                    "analyzer": "lucene.standard",
                    "name": "my_synonyms",
                    "source": {
                        "collection": "synonymous_terms"
                    }
                }
            ]
        }
    }

    // Create the index
    const result = await collection.createSearchIndex(index);
    console.log("New index name: " + result);
  } finally {
    await client.close();
  }
}

run().catch(console.dir);